runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -79,15 +86,43 @@
 	@Inject
 	private GrandExchangeConfig config;
 
+	private static Map<String, Integer> geLimitInfo;
+
+
 	@Provides
 	GrandExchangeConfig provideConfig(ConfigManager configManager)
 	{
 		return configManager.getConfig(GrandExchangeConfig.class);
 	}
 
+	private Map<String, Integer> loadGeLdimits()
+	{
+		Gson gson = new Gson();
+		Type type = new TypeToken<Map<String, Integer>>()
+		{
+		}.getType();
+
+		InputStream limitsFile = GrandExchangePlugin.class.getResourceAsStream("/ge_limits.json");
+		return gson.fromJson(new InputStreamReader(limitsFile), type);
+	}
+
+	static double getLimit(String itemName)
+	{
+
+		if (!geLimitInfo.containsKey(itemName))
+		{
+			return -1;
+		}
+
+		return geLimitInfo.get(itemName);
+
+	}
+
 	@Override
 	protected void startUp() throws IOException
 	{
+		geLimitInfo = loadGeLdimits();
```

Comment:
Setting up a static field in the instance's `startUp` method seems like not the best way to do this. Unfortunately I'm not entirely sure what the best alternative approach would be here. The only two that come to mind are: use a static initializer to set up the Map, or, scrap all the JSON stuff and add a `BuyLimit` enum, (with a method for lookups) instead